lib/excon/connection.rb

```diff
@@ -155,6 +155,8 @@ def request_call(datum)
                 break
               end
             end
+          elsif body.nil?
+            socket.write(request) # write out request + headers
           elsif !body.nil? # write out body
```

Comment:
Shouldn't break anything as elsif, but yeah, else seems easier/clearer. I've fixed it here: https://github.com/excon/excon/commit/24602e2316f0b08734eff4ff72db81662cb26126